src/observers/form_link_interceptor.ts

```diff
@@ -0,0 +1,52 @@
+import { LinkInterceptor, LinkInterceptorDelegate } from "../core/frames/link_interceptor"
+
+export type FormLinkInterceptorDelegate = {
+  shouldInterceptFormLinkClick(link: Element): boolean
+  formLinkClickIntercepted(link: Element, form: HTMLFormElement): void
+}
+
+export class FormLinkInterceptor implements LinkInterceptorDelegate {
+  readonly linkInterceptor: LinkInterceptor
+  readonly delegate: FormLinkInterceptorDelegate
+
+  constructor(delegate: FormLinkInterceptorDelegate, element: HTMLElement) {
+    this.delegate = delegate
+    this.linkInterceptor = new LinkInterceptor(this, element)
+  }
+
+  start() {
+    this.linkInterceptor.start()
+  }
+
+  stop() {
+    this.linkInterceptor.stop()
+  }
+
+  shouldInterceptLinkClick(link: Element): boolean {
+    return (
+      this.delegate.shouldInterceptFormLinkClick(link) &&
+      (link.hasAttribute("data-turbo-method") || link.hasAttribute("data-turbo-stream"))
+    )
+  }
+
+  linkClickIntercepted(link: Element, action: string): void {
+    const form = document.createElement("form")
+    form.setAttribute("action", action)
+    form.hidden = true
```

Comment:
Apologies if this edge case is already handled, I don't have a full understanding of all of the internals of Turbo.

But in the case that `Turbo.session.drive = false`, but on this link `data-turbo="true"` is specified, then wouldn't the form created by this also need to set `data-turbo="true"` on the form element for it to be handled by Turbo?

I have a PR open to address this (#500) but the addition of this FormLinkInterceptor would change where that behavior should be implemented.

Reply:
Thank you for raising these concerns! I've pushed up https://github.com/hotwired/turbo/pull/631/commits/d317319dadb5825f710009666cf6b27785839618 to add test coverage for the cases you've outlined.